File: algo/80. Remove Duplicates from Sorted Array II.py
# Two Pointers

class Solution:
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        limit = 0
        left = -1
        for i, num in enumerate(nums):
            if (left >= 0):
                if num == nums[left] and limit == 2:
                    continue
                elif num != nums[left]:
                    limit = 0
            left += 1
            nums[left] = num
            limit += 1
        return left + 1
        

File: round3/80. Remove Duplicates from Sorted Array II.cs
public class Solution {
    public int RemoveDuplicates(int[] nums) {
        int count = 1;
        int cursor = 1;
        
        for (int i = 1; i < nums.Length; i ++) {
            if (nums[i] == nums[i-1] && count == 2) {
                continue;
            } else {
                if (nums[i] != nums[i-1]) count = 0;
                nums[cursor ++] = nums[i];                
                count ++;
            }
        }
        
        return cursor;
    }
}